### PR TITLE
Shells/mocks for Kinesis Auto setup API endpoints

### PR DESCRIPTION
**Pull request by danotorrey**
This PR contains mocked out API endpoints for the Kinesis auto setup.

1) 3 individual step API endpoints to perform the automatic setup one step at a time.
2) 1 full_setup API endpoint.

_All of these API endpoints are 100% mocked. None of them actually interact with AWS, so you can execute them as frequently as you like._

The System > Nodes > API Browser has the full listing of routes and payloads.

#165 has a list of all steps and the actions they perform in detail. Also see the new [KinesisSetupResource](https://github.com/Graylog2/graylog-plugin-integrations/blob/2cdb8dbfd5bac5f997233cc0d00236cfea7c327a/src/main/java/org/graylog/integrations/aws/resources/KinesisSetupResource.java), which contains the Java definition for the endpoints and comments for each method.

The best place to get the post body for each request is the API browser. @ceruleancee can help with how each field should be presented in the UI. 

Note that all the API Browser displays the steps out of order, so we've added a *"Step X:"* before each of the individual steps.
image

Here's a sample of what the Full Setup request looks like:
image

### CODE REVIEW

File: src/main/java/org/graylog/integrations/IntegrationsModule.java [hunk 1]
```diff
@@ -88,6 +89,7 @@ protected void configure() {
         addMessageInput(AWSInput.class);
         addPermissions(AWSPermissions.class);
         addRestResource(AWSResource.class);
+        addRestResource(KinesisSetupResource.class);
         addTransport(AWSTransport.NAME, AWSTransport .class);
```
Comment (ceruleancee) [inline src/main/java/org/graylog/integrations/IntegrationsModule.java]: delete extra space

File: src/main/java/org/graylog/integrations/aws/resources/AWSResource.java [hunk 2]
```diff
@@ -128,17 +128,6 @@ public Response kinesisHealthCheck(@ApiParam(name = "JSON body", required = true
         return Response.accepted().entity(response).build();
```
Comment (ceruleancee) [inline src/main/java/org/graylog/integrations/aws/resources/AWSResource.java]: delete unneeded imports
import org.graylog.integrations.aws.resources.requests.KinesisNewStreamRequest;
import org.graylog.integrations.aws.resources.responses.KinesisNewStreamResponse;

File: src/main/java/org/graylog/integrations/aws/resources/KinesisSetupResource.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
+package org.graylog.integrations.aws.resources;
+
+import com.codahale.metrics.annotation.Timed;
+import io.netty.handler.codec.http.FullHttpRequest;
```
Comment (ceruleancee) [inline src/main/java/org/graylog/integrations/aws/resources/KinesisSetupResource.java]: delete import

File: src/main/java/org/graylog/integrations/aws/resources/KinesisSetupResource.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import org.apache.shiro.authz.annotation.RequiresAuthentication;
+import org.apache.shiro.authz.annotation.RequiresPermissions;
+import org.graylog.integrations.aws.AWSPermissions;
+import org.graylog.integrations.aws.resources.requests.AWSRequestImpl;
```
Comment (ceruleancee) [inline src/main/java/org/graylog/integrations/aws/resources/KinesisSetupResource.java]: same thing here, delete unneeded import

File: src/main/java/org/graylog/integrations/aws/resources/KinesisSetupResource.java [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+import org.graylog.integrations.aws.resources.requests.CreateLogSubscriptionPolicyRequest;
+import org.graylog.integrations.aws.resources.requests.CreateLogSubscriptionRequest;
+import org.graylog.integrations.aws.resources.requests.KinesisFullSetupRequest;
+import org.graylog.integrations.aws.resources.requests.KinesisNewStreamRequest;
+import org.graylog.integrations.aws.resources.responses.CreateLogSubscriptionPolicyResponse;
+import org.graylog.integrations.aws.resources.responses.CreateLogSubscriptionResponse;
+import org.graylog.integrations.aws.resources.responses.KinesisFullSetupResponse;
+import org.graylog.integrations.aws.resources.responses.KinesisFullSetupResponseStep;
+import org.graylog.integrations.aws.resources.responses.KinesisNewStreamResponse;
+import org.graylog.integrations.aws.resources.responses.LogGroupsResponse;
```
Comment (ceruleancee) [inline src/main/java/org/graylog/integrations/aws/resources/KinesisSetupResource.java]: and also here, delete unneeded import

File: src/main/java/org/graylog/integrations/aws/resources/requests/CreateLogSubscriptionRequest.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+@JsonAutoDetect
+@AutoValue
+@WithBeanGetter
+public abstract class CreateLogSubscriptionRequest implements AWSRequest {
+
+    private static final String LOG_GROUP_NAME = "log_group_name";
+    private static final String FILTER_NAME = "filter_name";
+    private static final String FILTER_PATTERN = "filter_pattern";
+    private static final String DESTINATION_STREAM_ARN = "destination-stream-arn";
```
Comment (ceruleancee) [inline src/main/java/org/graylog/integrations/aws/resources/requests/CreateLogSubscriptionRequest.java]: best to keep format consistent, `destination_stream_arn`

File: src/main/java/org/graylog/integrations/aws/resources/requests/CreateLogSubscriptionRequest.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    private static final String ROLE_ARN = "role-arn";
```
Comment (ceruleancee) [inline src/main/java/org/graylog/integrations/aws/resources/requests/CreateLogSubscriptionRequest.java]: best to keep format consistent, `destination_stream_arn` and `role-arn`

File: src/main/java/org/graylog/integrations/aws/resources/requests/CreateLogSubscriptionPolicyRequest.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+import org.graylog.autovalue.WithBeanGetter;
+
+@JsonAutoDetect
+@AutoValue
+@WithBeanGetter
+public abstract class CreateLogSubscriptionPolicyRequest implements AWSRequest {
+
+    private static final String STREAM_NAME = "stream_name";
+    private static final String STREAM_ARN = "stream_arn";
+    private static final String ROLE_NAME = "role_name";
```
Comment (ceruleancee) [inline src/main/java/org/graylog/integrations/aws/resources/requests/CreateLogSubscriptionPolicyRequest.java]: this is more of a preference, but `ROLE_NAME` could be listed first above `STREAM_NAME`, to keep consistent with the rest of the class

File: src/main/java/org/graylog/integrations/aws/resources/requests/KinesisFullSetupRequest.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+import org.graylog.autovalue.WithBeanGetter;
+
+@JsonAutoDetect
+@AutoValue
+@WithBeanGetter
+public abstract class KinesisFullSetupRequest implements AWSRequest {
+
+    private static final String LOG_GROUP_NAME = "log_group_name";
+    private static final String STREAM_NAME = "stream_name";
+    private static final String ROLE_NAME = "role_name";
```
Comment (ceruleancee) [inline src/main/java/org/graylog/integrations/aws/resources/requests/KinesisFullSetupRequest.java]: again a preference, `ROLE_NAME` could be listed first

File: src/main/java/org/graylog/integrations/aws/resources/responses/CreateLogSubscriptionPolicyResponse.java [hunk 10]
```diff
@@ -0,0 +1,27 @@
+package org.graylog.integrations.aws.resources.responses;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
```
Comment (ceruleancee) [inline src/main/java/org/graylog/integrations/aws/resources/responses/CreateLogSubscriptionPolicyResponse.java]: delete unneeded import

File: src/main/java/org/graylog/integrations/aws/resources/responses/KinesisFullSetupResponse.java [hunk 11]
```diff
@@ -0,0 +1,34 @@
+package org.graylog.integrations.aws.resources.responses;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
```
Comment (ceruleancee) [inline src/main/java/org/graylog/integrations/aws/resources/responses/KinesisFullSetupResponse.java]: delete unneeded import

File: src/main/java/org/graylog/integrations/aws/resources/responses/KinesisFullSetupResponseStep.java [hunk 12]
```diff
@@ -0,0 +1,33 @@
+package org.graylog.integrations.aws.resources.responses;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.auto.value.AutoValue;
+import org.graylog.autovalue.WithBeanGetter;
+
+import java.util.List;
```
Comment (ceruleancee) [inline src/main/java/org/graylog/integrations/aws/resources/responses/KinesisFullSetupResponseStep.java]: delete uneeded import

File: src/main/java/org/graylog/integrations/aws/resources/requests/CreateLogSubscriptionPolicyRequest.java (hunk 8 above)
Comment (danotorrey) [reply src/main/java/org/graylog/integrations/aws/resources/requests/CreateLogSubscriptionPolicyRequest.java]: Great catch! Agreed.

**Comment by danotorrey** [comment]
@ceruleancee If this PR looks good to you when reviewing again, please go ahead and merge it.

**Pull request merged**